R/globals.R

```diff
@@ -25,6 +25,28 @@ register_s3_method <- function(pkg, generic, class, fun = NULL) {
   )
 }
 
+register_upgrade_message <- function(pkg, version) {
+
+  msg <- sprintf(
+    "This version of Shiny is designed to work with '%s' >= %s.
+    Please upgrade via install.packages('%s').",
+    pkg, version, pkg
+  )
+
+  # emit startup message of the package is already loaded
+  if (pkg %in% loadedNamespaces()) {
```

Comment:
Don't you need to check the version number?

Reply:
whoops, yep